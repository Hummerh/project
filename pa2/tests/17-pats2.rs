fn main() {
      let mut y : i32 = -3;

      match (y) {
            1 | 2 | 3 => { },
            -3 => { },
            _ => { }
      }
}
