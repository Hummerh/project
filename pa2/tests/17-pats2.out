(crate
    (items
        (fn-def
            (id
                (main))
            (block
                (let
                    (pat-mut-id
                        (id
                            (y)))
                    (type-i32)
                    (neg
                        (lit-dec)))
                (match
                    (id
                        (y))
                    (match-arms
                        (match-arm
                            (pats
                                (pat-lit
                                    (lit-dec))
                                (pat-lit
                                    (lit-dec))
                                (pat-lit
                                    (lit-dec)))
                            (block))
                        (match-arm
                            (pats
                                (pat-lit
                                    (lit-dec)))
                            (block))
                        (match-arm
                            (pats
                                (pat-wild))
                            (block))))))))